tests/Feature/Api/Assets/AssetStoreTest.php

```diff
@@ -482,35 +482,52 @@ public function testCompanyIdNeedsToBeInteger()
             });
     }
 
-    public function testEncryptedCustomField()
+    public function testEncryptedCustomFieldCanBeStored()
     {
+        $status = Statuslabel::factory()->create();
         $field = CustomField::factory()->testEncrypted()->create();
-        $asset = Asset::factory()->has_encrypted_custom_field()->create();
         $superuser = User::factory()->superuser()->create();
-        $normal_user = User::factory()->editAssets()->create();
+        $assetData = Asset::factory()->hasEncryptedCustomField($field)->make();
 
-        //first, test that an Admin user can save the encrypted custom field
         $response = $this->actingAsForApi($superuser)
-            ->patchJson(route('api.assets.update', $asset->id), [
-                $field->db_column_name() => 'This is encrypted field'
+            ->postJson(route('api.assets.store'), [
+                $field->db_column_name() => 'This is encrypted field',
+                'model_id' => $assetData->model->id,
+                'status_id' => $status->id,
+                'asset_tag' => '1234',
             ])
             ->assertStatusMessageIs('success')
             ->assertOk()
             ->json();
-        $asset->refresh();
-        $this->assertEquals(\Crypt::decrypt($asset->{$field->db_column_name()}), 'This is encrypted field');
 
-        //next, test that a 'normal' user *cannot* change the encrypted custom field
+        $asset = Asset::findOrFail($response['payload']['id']);
+        $this->assertEquals('This is encrypted field', Crypt::decrypt($asset->{$field->db_column_name()}));
+    }
+
+    public function testPermissionNeededToStoreEncryptedField()
+    {
+        // @todo:
+        $this->markTestIncomplete();
+
+        $status = Statuslabel::factory()->create();
+        $field = CustomField::factory()->testEncrypted()->create();
+        $normal_user = User::factory()->editAssets()->create();
+        $assetData = Asset::factory()->hasEncryptedCustomField($field)->make();
+
         $response = $this->actingAsForApi($normal_user)
-            ->patchJson(route('api.assets.update', $asset->id), [
-                $field->db_column_name() => 'Some Other Value Entirely!'
+            ->postJson(route('api.assets.store'), [
```

Comment:
This Patch -> Post thing, is that important? Or is it pretty much the same thing? Or are we talking about, maybe, two different things - updating an existing thing versus making a new thing?